qiskit/synthesis/su4/utils.py

```diff
@@ -0,0 +1,88 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2023
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+# pylint: disable=invalid-name
+
+"""Utilities for SU(4) synthesis: Weyl chamber, polytopes, distances, etc."""
+
+import numpy as np
+
+
+_EPS = 1e-12
+
+
+def find_min_point(P):
+    """Find the closest point on convex polytope P to the origin (in Euclidean distance).
+
+    P is given as a list of points forming its vertices.
+
+    Function from
+    https://scipy-cookbook.readthedocs.io/items/Finding_Convex_Hull_Minimum_Point.html
+    """
+    if len(P) == 1:
+        return P[0]
+
+    P = [np.array(p) for p in P]
```

Comment:
You probably mean `np.asarray` (everywhere in this function), since you're not trying to mutate things or need contiguous memory, and you probably don't need a copy made if things are already an array.